### PR TITLE
Create movie pagination and filter

### PR DESCRIPTION
**Pull request by jgoh120**
create pagination and filter based on avg rating and genre choice.

### CODE REVIEW

File: routers/movie.router.js [hunk 1]
```diff
@@ -5,8 +5,20 @@ const auth = require('../config/auth').auth;
...
+    // res.json(movies);
+    const pagination = {
+        sortBy: _.get(req.query, 'sortBy', 'createdAt'), // createdAt, rating
+        direction: _.get(req.query, 'direction', 'desc'), // desc, asc
+        limit: parseInt(_.get(req.query, 'limit', '3')), // any number > 0
+        page: parseInt(_.get(req.query, 'page', '1')), // any number >= 1
+    };
+    const filter = {
+        averageRating: parseInt(_.get(req.query, 'filterRating', '0')),
+        genre: _.get(req.query, 'filterGenre', null)
```
Comment (clarencecastillo) [inline routers/movie.router.js:18]: We expect a value like `horror,comedy` from the request query. Hence, we'll need to convert the string value into an array like so:
`genres: _.get(req.query, 'filterGenres', '').split(',')`

**Review by clarencecastillo — CHANGES_REQUESTED** [review]

File: controllers/movie.controller.js [hunk 2]
```diff
@@ -16,9 +16,56 @@ class MovieController {
...
 
-    async getAll() {
-        const movies = await this.movieRepository.find();
-        return movies.map(m => this.formatMovie(m));
+    // async getAll() {
+    //     const movies = await this.movieRepository.find();
+    //     return movies.map(m => this.formatMovie(m));
+    // }
+
+    async getPageOfChoice(pagination, filter){
```
[the remark this answers is not in the document]
Comment (jgoh120) [reply controllers/movie.controller.js:24]: Done

**Comment by jgoh120** [comment]
all done

File: controllers/movie.controller.js [hunk 3]
```diff
@@ -16,9 +16,52 @@ class MovieController {
...
+            if (filter.rating > 0 && filter.rating <= 5) {
+                query.rating = filter.rating;
+            }
+            if (filter.genre != null) {
+                query['genre'] = filter.genre;
+            }
+        }
+        
+        const movies = await this.movieRepository.find(query, null, queryOptions);
+        return movies.map(r => this.formatMovie);
```
Comment (clarencecastillo) [inline controllers/movie.controller.js:64]: Rename local variable `r` into `m` and pass it as argument to `formatMovie` method.

File: routers/movie.router.js [hunk 4]
```diff
@@ -5,8 +5,18 @@ const auth = require('../config/auth').auth;
 var reviewRouter = require('./review.router');
```
Comment (clarencecastillo) [inline routers/movie.router.js:5]: We need to import lodash first before we can use it:
`const _ = require('lodash');`

File: controllers/movie.controller.js [hunk 5]
```diff
@@ -16,9 +16,52 @@ class MovieController {
...
+    }
+
+    async getTotalCount(filter){
+        const query = {};
+        if (filter != undefined) {
+            if (filter.rating > 0 && filter.rating <= 5) {
+                query.rating = filter.rating;
+            }
+
+            if (filter.genres != null) {
```
Comment (clarencecastillo) [inline controllers/movie.controller.js:37]: Rename the attribute `genres` to be just `genre` according to schema definition.

**Review by clarencecastillo — CHANGES_REQUESTED** [review]

**Review by clarencecastillo — APPROVED** [review]

**Pull request merged**